### PR TITLE
Core 1282 login redirect

### PR DESCRIPTION
**Pull request by kare**

### CODE REVIEW

File: grails-app/controllers/com/unifina/controller/security/LoginController.groovy [hunk 1]
```diff
@@ -54,8 +52,19 @@ class LoginController {
...
 		String postUrl = "${request.contextPath}${config.apf.filterProcessesUrl}"
-		render view: view, model: [postUrl: postUrl,
-		                           rememberMeParameter: config.rememberMe.parameter]
+		def redirect = params.redirect
+		if (redirect != null) {
+			redirect = redirect.trim()
+			// Match https://www.streamr.com/ with required subdomain.
+			// Subdomain can be anything within [a-zA-Z0-9]
+			// Note required trailing slash!
+			if (!redirect.matches("^https://([a-zA-Z0-9]*)?\\.streamr\\.com/.*")) {
```
Comment (harbu) [inline grails-app/controllers/com/unifina/controller/security/LoginController.groovy]: We should also support `http://localhost:3333`, `http://127.0.0.1:3333` (or just `http://localhost:<ANYPORT>` and `http://127.0.0.1:<ANYPORT>` to future-proof any port changes)

Comment (kare) [reply grails-app/controllers/com/unifina/controller/security/LoginController.groovy]: Done

File: grails-app/controllers/com/unifina/controller/security/LoginController.groovy [hunk 2]
```diff
@@ -55,10 +55,7 @@ class LoginController {
 		def redirect = params.redirect
 		if (redirect != null) {
 			redirect = redirect.trim()
-			// Match https://www.streamr.com/ with required subdomain.
-			// Subdomain can be anything within [a-zA-Z0-9]
-			// Note required trailing slash!
-			if (!redirect.matches("^https://([a-zA-Z0-9]*)?\\.streamr\\.com/.*")) {
+			if (!LoginRedirectValidator.isValid(redirect)) {
```
Comment (harbu) [inline grails-app/controllers/com/unifina/controller/security/LoginController.groovy]: Good call, encapsulating the logic!

File: test/unit/com/unifina/controller/security/LoginRedirectValidatorSpec.groovy [hunk 3]
```diff
@@ -0,0 +1,30 @@
+package com.unifina.controller.security
+
+import spock.lang.Specification
+
+class LoginRedirectValidatorSpec extends Specification {
```
Comment (harbu) [inline test/unit/com/unifina/controller/security/LoginRedirectValidatorSpec.groovy]: Nice one

**Pull request merged**